adhocteam/Head-Start-TTADP
# Cm 60 reconcile legacy reports

**Description of change**
Add the ability to reconcile email addresses for author, approving manager, and collaborators between legacy reports and current users.

This is the first of two pull request on this issue. This adds the reconciliation code and the tooling to run reconciliations manually. The next pull request will add a recurring nightly job to the queue to be executed automatically by the worker process.

**How to test**
1. Pull down changes and run `yarn docker:reset`
2. Start the application `yarn docker:start`
3. upload `R14ActivityReportsTest.csv` to minio
4. import the legacy report `yarn import:reports:local --file R14ActivityReportsTest.csv --region 1`
3. Run `yarn reconcile:legacy`
4. Nothing should change with the reports
5. Change a user's email (ex. Harry potter) to specialist1@test.gov
6. rerun `yarn reconcile:legacy`
7. You should see log messages about changing the author and adding collaborators for reports.
7. You should see that user as the author of a few reports and a collaborator on a couple more.
8. Change two more user's emails to be specialist2@test.gov and specialist3@test.gov.
9. rerun `yarn reconcile:legacy`
10. You should see log messages about changing the author and adding collaborators for reports.
10. Now all reports should have authors and four should have collaborators.
11. Change a final user's email address to be manager@test.gov
12. rerun 'yarn reconcile:legacy'
13. You should see log messages about changing the approving manager for reports.
14. If you examine the DB, you should see all legacy reports now have authors and approving  managers.
15. You should be able to rerun `yarn reconcile:legacy` multiple times with nothing changing.


**Issue(s)**
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-60

**Checklist**

    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
